### PR TITLE
refactor: split off blockExit into separate module blockexit.d

### PR DESCRIPTION
**Pull request by WalterBright**
This was always awkward having a Visitor nested inside another Visitor. There's no reason for it to be there - it doesn't rely on anything in the uplevel scopes. Making a new module out of it does a much better job of encapsulating it.

### CODE REVIEW

**Review by UplinkCoder — CHANGES_REQUESTED** [review]
Define the BE-enum in `blockexit.d` as well please

**Comment by UplinkCoder** [comment]
Otherwise I really like this change.

**Comment by WalterBright** [comment]
I plan on doing more than that in the future. The first step is just separating it to make it more tractable.

File: src/statement.d [hunk 1]
```diff
@@ -241,491 +236,13 @@ extern (C++) abstract class Statement : RootObject
     }
 
     /* ============================================== */
-    /** Only valid after semantic analysis
```
Comment (UplinkCoder) [inline src/statement.d]: Please add the second * back in.
So the IDE shows it.

Comment (wilzbach) [inline src/statement.d]: FWIW at some point there was also the movement to expose `dmd` as library and there only `**` comments would be publicly documented.

File: src/blockexit.d [hunk 2]
```diff
@@ -0,0 +1,503 @@
...
+import ddmd.globals;
+import ddmd.id;
+import ddmd.identifier;
+import ddmd.mtype;
+import ddmd.statement;
+import ddmd.tokens;
+import ddmd.visitor;
+
+
+/* ============================================== */
```
Comment (UplinkCoder) [inline src/blockexit.d]: add a * here

File: src/statement.d (hunk 1 above)
Comment (WalterBright) [inline src/statement.d]: Ah, that's why it disappeared.

**Review by UplinkCoder — APPROVED** [review]

**Pull request merged**